test/robot/go_robot_go.sh

```diff
@@ -0,0 +1,320 @@
+#!/bin/bash
+shopt -s extglob
+clear
+pushd $(dirname $0/../..) > /dev/null
+OUTDIR="bazel-out/robot/"
+BUILD_BOT_DIR="${OUTDIR}build_bot/"
+
+function handle_opts() {
+  while true; do
+    case $1 in
+      -c|--clean) CLEAN=1; shift ;;
+      --) shift; break ;;
+      *) echo "internal error!" ; exit 1 ;;
+    esac
+  done
+}
+handle_opts $(getopt -l "clean" -o "c" -n "go_robot_go.sh" -- "$@")
+
+function howmany() { echo $#; }
+
+function find_pid() { pgrep -f "$1" -d " "; }
+
+function is_device_awake() {
+  STATE=`adb -s $1 shell dumpsys nfc \
+    | grep mScreenState \
+    | awk '{split($1,a,"="); print a[2]}'`
+
+  if [[ "$STATE" = "ON_UNLOCKED" ]]; then
+    echo 0
+    return
+  fi
+  echo 1
+  return
+}
+
+function start_swipe() {
+  SWIPE_COMMAND="watch -n60 adb -s"
+  OLD_SWIPE_PID=$(find_pid "$SWIPE_COMMAND")
+  if [[ $OLD_SWIPE_PID ]]; then
+    echo ">>> Found old swipe process: ${OLD_SWIPE_PID}; Killing..."
+    kill -9 $OLD_SWIPE_PID
+  fi
+
+  echo ">>> Starting swipe process..."
+  let NUM_DEVICES=0
+  for device in $(adb devices | grep -v List | awk '{print $1}'); do
+    if [[ $(is_device_awake $device) -eq 0 ]]; then
+      echo ">>> starting swipe process on $device"
+      nohup watch -n60 "adb -s $device shell input swipe 300 300 0 0 500" > /dev/null 2>&1 &
+      let NUM_DEVICES++
+    fi
+  done
+  # need to sleep, else we get duplicate processes in our pid list from watch
+  sleep 2s
+  SWIPE_PID=$(find_pid "$SWIPE_COMMAND")
+  echo ">>> Swiping processes started! PIDs: ${SWIPE_PID}"
+  if [[ $(howmany ${SWIPE_PID}) -ne ${NUM_DEVICES} ]]; then
+    echo ">>> Warning, swiping processed started does not match devices connected..."
+    echo "    processes found: ($(howmany ${SWIPE_PID}))"
+    echo "    number of devices: (${NUM_DEVICES})"
+  fi
+}
+
+function end_swipe() {
+  for spid in ${SWIPE_PID}; do
```

Comment:
I don't really like that this uses a variable that gets defined in another function.